src/material/icon/icon.scss

```diff
@@ -59,6 +60,33 @@ mat-icon {
 
     .mat-icon-button .mat-icon {
       margin: auto;
+      overflow: visible;
     }
   }
 }
+
+// Makes button icon overflow visible to fix b/250063405
+.mat-mdc-icon-button .mat-icon,
+.mdc-button .mat-icon,
+.mat-mdc-fab-base .mat-icon,
+.mat-toolbar .mat-icon,
+.mat-mdc-tooltip-trigger .mat-icon,
+.mat-mdc-tab .mat-icon,
+.mat-mdc-form-field .mat-icon,
+.mat-grid-tile .mat-icon,
+.mat-mdc-list-item .mat-icon,
+.mat-mdc-list-option .mat-icon {
+  overflow: visible;
+  // Apply styles to avoid text spacing
+  letter-spacing: 0;
+  word-spacing: 0;
+  line-height: 1;
+}
+
+// Makes button icon visibility hidden to fix b/250063405 to avoid
+// overrunning into sibling/neighboring elements.
+@media (variables.$xsmall) {
```

Comment:
I don't think this is an assumption we can make. Even if the screen is small, there can be enough space to show the icon.

Reply:
Hi @crisbeto! Removed and ended up making updates to .mdc-button min-width/min-height in latest change [46dcaf2](https://github.com/angular/components/pull/29686/commits/46dcaf25e304f510fedf39510c49e804dd1d9723).

[Before Screenshot](https://screenshot.googleplex.com/99fiTHt2JPr6B8U)
[Updated AFTER Screenshot](https://screenshot.googleplex.com/8pYHWmwtAGc9baw)

Let me know if this is acceptable.